code/api.tf

```diff
@@ -10,6 +10,7 @@ resource "azurerm_container_registry" "container_registry" {
   resource_group_name = azurerm_resource_group.rg_api.name
   location            = azurerm_resource_group.rg_api.location
   sku                 = var.sku_container_registry
+  admin_enabled       = true
```

Comment:
@ElfoLiNk why do you need to enable the admin user?
If so don't you have to define outputs to show admin user and its password?
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_registry#admin_username

Reply:
app_service needs authorization to be able to pull from the registry:

```
Failed to load ACR Credentials needed to load Images and Tags - Cannot perform credential operations for /subscriptions/bc25e994-b666-4414-8ca7-2769a394008b/resourceGroups/cgnportaltest-d-api-rg/providers/Microsoft.ContainerRegistry/registries/cgnportaltestdarc as admin user is disabled. Kindly enable admin user as per docs: https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication#admin-account.
```

```
2021-04-15T13:04:25.681Z INFO  - Pulling image from Docker hub: cgnportaltestdarc.azurecr.io/cgn-onboarding-portal-backend:latest
2021-04-15T13:04:25.882Z ERROR - DockerApiException: Docker API responded with status code=InternalServerError, response={"message":"Get https://cgnportaltestdarc.azurecr.io/v2/cgn-onboarding-portal-backend/manifests/latest: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information."}

2021-04-15T13:04:25.890Z ERROR - Image pull failed: Verify docker image configuration and credentials (if using private repository)
2021-04-15T13:04:33.675Z INFO  - Stopping site cgnportaltest-d-portal-backend1 because it failed during startup.
```

One solution is to give the app_service principal the AcrPull authorization role with

https://github.com/pagopa/cgn-onboarding-portal-infra/blob/41c727edfa11ad8109da6313699ec5779ffe73aa/code/api.tf#L27-L33

but the terraform service principal needs further permission not available with only the Contributor role.

The other one was to pass the registry admin credential to the app_service in the app_settings https://github.com/pagopa/cgn-onboarding-portal-infra/blob/41c727edfa11ad8109da6313699ec5779ffe73aa/code/api.tf#L49-L50

Reply:
@ElfoLiNk get it. Thank you ...